app/src/main/java/org/dslul/openboard/inputmethod/keyboard/MainKeyboardView.java

```diff
@@ -207,7 +207,12 @@ public MainKeyboardView(final Context context, final AttributeSet attrs, final i
                 LANGUAGE_ON_SPACEBAR_TEXT_SHADOW_RADIUS_DISABLED);
         mLanguageOnSpacebarTextShadowColor = mainKeyboardViewAttr.getColor(
                 R.styleable.MainKeyboardView_languageOnSpacebarTextShadowColor, 0);
-        mLanguageOnSpacebarFinalAlpha = Color.alpha(mLanguageOnSpacebarTextColor);
+        // Add the alpha value here for fading dynamic colors, as system_accent or system_neutral colors have no alpha value.
+        if (colors.getThemeColors().equals(KeyboardTheme.THEME_DYNAMIC_LIGHT) || colors.getThemeColors().equals(KeyboardTheme.THEME_DYNAMIC_DARK)) {
+            mLanguageOnSpacebarFinalAlpha = Color.alpha(applyAlpha(mLanguageOnSpacebarTextColor, 0.5f));
```

Comment:
Why don't you modify the color directly when creating the `Colors` instance? It looks like you set the spacebar text to a color you don't actually want to use, and then modify it every time it's applied.

Reply:
I've noticed that this effect is applied in this part of the code and not elsewhere.
That's why I thought it would be a good idea to add this transparency here for the spacebar text color...

So it would be better if :
- In the "ColorUtil.kt" file, I add the "applyAlpha" function;
- In the "Colors.kt" file, I set the color of the spacebar text here?
With these changes, it also works...

Reply:
My idea would be setting the the color directly. So in `KeyboardTheme` instead of `ContextCompat.getColor(context, R.color.text_color_dynamic_light)` you set something like `ColorUtils.setAlphaComponent(ContextCompat.getColor(context, R.color.text_color_dynamic_light), 127)`

You can also use `applyAlpha` instead, but then it should be in `ColorUtil`.

Reply:
I didn't know I could do it like that. 😅
Solved in a06e4ff